build-tools/packages/version-tools/src/internalVersionScheme.ts

```diff
@@ -497,3 +497,47 @@ export function detectInternalVersionConstraintType(
 	const maxSatisfying = semver.maxSatisfying([patch, minor], range);
 	return maxSatisfying === patch ? "patch" : maxSatisfying === minor ? "minor" : "exact";
 }
+
+/**
+ * Checks if the provided version is a test version.
+ *
+ * Test versions are generated from test/ branches and are published to the test feed.
+ *
+ * @param version - The version to check
+ * @returns - True if the version string is a test version, otherwise false
```

Comment:
And apparently throws in some situations. Document?

Reply:
I don't think you want to both throw and return. I can't remember which functions exactly, but elsewhere there is a parse function that throws, and the "is" function calls the parse one and handles the errors. That way the is function purely returns booleans. Not sure it's worth it in this case but imo this shouldn't throw.

Reply:
I haven't been able to find the function you mentioned. I see assert or throw being used to check if `parsedVersion` is null. happy to follow-up on this.